File: Samples/PISystem/CSharp/GSharp.PythonRegressionAnalysis/GSharp.Testbed/Program.Init.cs
using GHost.Core.Logging;
using GHost.GSharp.Core;
using GHost.GSharp.Core.Enums;
using GHost.GSharp.Developer;
using System;
using System.Diagnostics;

namespace GSharp.Testbed;

internal partial class Program
{
    private static void RunTestbed()
    {
        new Program().RunDebuggingHost();
    }

    private void RunDebuggingHost()
    {
        ILogger log = null;

        try
        {
            GlobalDataProvider.Initialize();

            var host = new AssemblyTestHost();

            ConfigureAssembly(host.GetBuilder());
            ConfigureSettings(host.Settings);

            log = host.Log;

            host.Start();
        }
        catch (Exception ex)
        {
            if (log is null)
                Console.WriteLine(ex.ToString());
            else
                log.Error(ex);
        }

        if (Debugger.IsAttached)
        {
            Console.Write("Press any key to continue . . . ");
            _ = Console.ReadKey(intercept: true);
        }
    }

    partial void ConfigureSettings(AssemblyTestHost.TestHostSettings settings);
}

public class GlobalDataProvider : GlobalDataProviderBase
{
    public static GlobalDataProvider Instance = new GlobalDataProvider();

    public static void Initialize() => GlobalData.GlobalDataProvider = Instance;


    public override string ProductTitle => "gSharp Developer Testbed";

    public override string ProductVersionString => Versions.ProductVersion;

    public override string LogSource => "$DevTestbed";

    public override string ProductName => "GHost.GSharp.Developer.Testbed";

    public override ProgramType ProgramType => ProgramType.DeveloperTestbed;

    public override string CompanyDataFolder => SolutionConstants.CompanyDataFolder;
}

File: Samples/PISystem/VB.NET/GSharp.PythonRegressionAnalysis/GSharp.Testbed/Program.Init.vb
Imports GHost.Core.Logging
Imports GHost.GSharp.Core
Imports GHost.GSharp.Core.Enums
Imports GHost.GSharp.Developer

Partial Friend Class Program

    Public Shared Sub RunTestbed()
        Dim prog As New Program()
        prog.RunDebuggingHost()
    End Sub

    Private Sub RunDebuggingHost()
        Dim Log As ILogger = Nothing

        Try
            GlobalDataProvider.Initialize()

            Dim host = New AssemblyTestHost()

            ConfigureAssembly(host.GetBuilder())
            ConfigureSettings(host.Settings)

            Log = host.Log

            host.Start()
        Catch ex As Exception
            If Log Is Nothing Then
                Console.WriteLine(ex.ToString())
            Else
                Log.Error(ex)
            End If
        End Try

        If Debugger.IsAttached Then
            Console.Write("Press any key to continue . . . ")
            Console.ReadKey(intercept:=True)
        End If
    End Sub

    Partial Private Sub ConfigureSettings(settings As AssemblyTestHost.TestHostSettings)
    End Sub

End Class

Public Class GlobalDataProvider
    Inherits GlobalDataProviderBase

    Public Shared Instance As New GlobalDataProvider()

    Public Shared Sub Initialize()
        GlobalData.GlobalDataProvider = Instance
    End Sub

    Public Overrides ReadOnly Property ProductTitle As String
        Get
            Return "gSharp Developer Testbed"
        End Get
    End Property

    Public Overrides ReadOnly Property ProductVersionString As String
        Get
            Return Versions.ProductVersion
        End Get
    End Property

    Public Overrides ReadOnly Property LogSource As String
        Get
            Return "$DevTestbed"
        End Get
    End Property

    Public Overrides ReadOnly Property ProductName As String
        Get
            Return "GHost.GSharp.Developer.Testbed"
        End Get
    End Property

    Public Overrides ReadOnly Property ProgramType As ProgramType
        Get
            Return ProgramType.DeveloperTestbed
        End Get
    End Property

    Public Overrides ReadOnly Property CompanyDataFolder As String
        Get
            Return SolutionConstants.CompanyDataFolder
        End Get
    End Property
End Class
